### ISSUE TITLE
For APIs, methods that don't return a value should not return a value in the interface stub

### ISSUE THREAD

**Issue by iislucas** [freedomjs/freedom#194]
Proposal: when a freedom API function is not intended to return something (has no return value in the json API spec), then the freedom stub interface to the module should also not return a value (rather than returning a promise, as currently happens).

**Comment by bemasc** [comment]
I think this actually fights with #194.  What happens if a void function throws an exception?  Returning a Promise allows us to indicate that to the caller.

However, this could be solved by rethrowing a matching exception ... or of course by ignoring this case.

**Comment by willscott** [comment]
Issue here is that we would lose the ability to provide a meaningful explanation of when the method returns on the client.

There are cases where you want to do something else after you know your request has taken effect, so despite lack of return value we have cases where there are ordering constraints need to be enforced. think about some of the WebRTC methods where we want to know that operation has finished before calling the next one.

**Comment by iislucas** [comment]
Yeah totally agree. I like the convention of being able to express return value of a void promise, no return value, or a return value within a promise. That seems like the 3 cases.
